src/us/kbase/userandjobstate/jobstate/JobState.java

```diff
@@ -716,34 +712,40 @@ public void deleteJob(final String user, final String jobID,
 		 */
 		checkString(user, "user");
 		auth.authorizeRead(strat, user, authParams);
-		String query;
+		final BasicDBObject query = startQuery(running, complete, canceled, error);
 		if (strat.equals(UJSAuthorizer.DEFAULT_AUTH_STRAT)) {
 			if (shared) {
-				query = String.format("{$or: [{%s: '%s'}, {%s: '%s'}]",
-						USER, user, SHARED, user);
+				// clean this up later
+				final List<BasicDBObject> shr = Arrays.asList(new BasicDBObject(USER, user),
+						new BasicDBObject(SHARED, user));
```

Comment:
What is the whole block doing here?

Reply:
It's setting up the authorization part of a query to list a user's jobs based on multifaceted criteria. In particular
- if the UJS is used for authorization, then it finds jobs owned by the user and optionally shared with the user
- if the WS is used for authorization, then it find jobs associated with a list of workspaces to which the user has read access
- it also does some hanky panky with the `query` object created by `startQuery` to not overwrite the `$or` key if there's one in there already.